internal/job/checkout.go

```diff
@@ -780,12 +781,21 @@ func gitFetchCommitWithFallback(ctx context.Context, shell *shell.Shell, gitFetc
 
 const CommitMetadataKey = "buildkite:git:commit"
 
+var potentiallyValidGitSHA = regexp.MustCompile(`^[a-f0-9]{40}|[a-f0-9]{64}$`)
+
 // sendCommitToBuildkite sends commit information (commit, author, subject, body) to Buildkite, as the BK backend doesn't
 // have access to user's VCSes. To do this, we set a special meta-data key in the build, but only if it isn't already present
 // Functionally, this means that the first job in a build (usually a pipeline upload or similar) will push the commit info
 // to buildkite, which uses this info to display commit info in the UI eg in the title for the build
 // note that we bail early if the key already exists, as we don't want to overwrite it
 func (e *Executor) sendCommitToBuildkite(ctx context.Context) error {
+	commitRef, _ := e.shell.Env.Get("BUILDKITE_COMMIT")
+	if commitRef != "HEAD" && potentiallyValidGitSHA.MatchString(commitRef) {
+		// we can skip the metadata shenanigans here and push straight through
```

Comment:
If `commitRef` matches the regex, then it can't possibly equal `"HEAD"`.

Reply:
Classic "writing the if statement as my train of thought was going". I'll remove the check.